src/Findologic/Config/FindologicConfigService.php

```diff
@@ -0,0 +1,307 @@
+            if ($id) {
+                $this->finSearchConfigRepository->delete([['id' => $id]], Context::createDefaultContext());
+            }
+
+            return;
+        }
+
+        $data = [
+            'id' => $id ?? Uuid::randomHex(),
+            'configurationKey' => $key,
+            'configurationValue' => $value,
+            'salesChannelId' => $salesChannelId,
+            'languageId' => $languageId,
+        ];
+        $this->finSearchConfigRepository->upsert([$data], Context::createDefaultContext());
+    }
+
+    public function delete(string $key, string $salesChannel, string $languageId): void
+    {
+        $this->set($key, null, $salesChannel, $languageId);
+    }
+
+    private function validate(string $key, string $salesChannelId, string $languageId): void
+    {
+        $key = trim($key);
+        if ($key === '') {
+            throw new InvalidKeyException('key may not be empty');
+        }
+        if ($salesChannelId && !Uuid::isValid($salesChannelId)) {
+            throw new InvalidUuidException($salesChannelId);
+        }
+        if ($languageId && !Uuid::isValid($languageId)) {
+            throw new InvalidUuidException($languageId);
+        }
+    }
+
+    private function getId(string $key, string $salesChannelId, string $languageId): ?string
+    {
+        $criteria = $this->buildCriteriaWithKey($key, $salesChannelId, $languageId);
+        $ids = $this->finSearchConfigRepository->searchIds($criteria, Context::createDefaultContext())->getIds();
+
+        return array_shift($ids);
+    }
+
+    private function buildCriteriaWithKey(string $key, string $salesChannelId, string $languageId): Criteria
+    {
+        $criteria = new Criteria();
+        $criteria->addFilter(
+            new EqualsFilter('configurationKey', $key),
+            new EqualsFilter('salesChannelId', $salesChannelId),
+            new EqualsFilter('languageId', $languageId)
+        );
+
+        return $criteria;
+    }
+
+    private function load(string $salesChannelId, string $languageId): array
+    {
+        $key = sprintf('%s-%s', $salesChannelId, $languageId);
+
+        if (isset($this->configs[$key])) {
+            return $this->configs[$key];
+        }
+
+        $criteria = new Criteria();
+        if (method_exists($criteria, 'setTitle')) {
+            $criteria->setTitle('finsearch-config::load');
+        }
+
+        $criteria->addFilter(
+            new MultiFilter(
+                MultiFilter::CONNECTION_AND,
+                [
+                    new EqualsFilter('salesChannelId', $salesChannelId),
+                    new EqualsFilter('languageId', $languageId),
+                ]
+            )
+        );
+
+        $criteria->addSorting(
+            new FieldSorting('salesChannelId', FieldSorting::ASCENDING),
+            new FieldSorting('id', FieldSorting::ASCENDING)
+        );
+        $criteria->setLimit(500);
+
+        $configs = new FinSearchConfigCollection();
```

Comment:
Why is this even declared if it is not used? In line 273 you explicitly use this collection, to save one configuration and in the next line you return it immediately. Is there a reason why it isn't returned directly?